sdk/appcenter-data/src/androidTest/java/com/microsoft/appcenter/data/LocalDocumentStorageAndroidTest.java

```diff
@@ -109,6 +109,45 @@ public void writeReadDelete() {
         assertNotNull(deletedDocument.getError());
     }
 
+    @Test
+    public void offlineUpdatePreservesEtag() {
+
+        /* Write document to the cache with an etag set. */
+        DocumentWrapper<String> document = new DocumentWrapper<>(TEST_VALUE, DefaultPartitions.USER_DOCUMENTS, ID, "etag", 0);
+        mLocalDocumentStorage.writeOnline(USER_TABLE_NAME, document, new WriteOptions());
+        DocumentWrapper<String> createdDocument = mLocalDocumentStorage.read(USER_TABLE_NAME, DefaultPartitions.USER_DOCUMENTS, ID, String.class, new ReadOptions());
+        assertNotNull(createdDocument.getETag());
+        assertEquals("etag", createdDocument.getETag());
+
+        /* Replace document offline, etag should be not replaced. */
```

Comment:
```suggestion
        /* Replace document offline, etag should not be replaced. */
```

That was the original suggestion ;) Can be applied from files tab directly on github as batch.

Reply:
Ah good to know. Should help prevent having a typo in typo fixes 😅